### PR TITLE
Fix pytest.fail() not captured in cucumber JSON step output

### PR DESCRIPTION
**Pull request by jackhenderson101**
pytest.fail() raises _pytest.outcomes.Failed which inherits from BaseException, not Exception. The except Exception handler in _execute_step_function was bypassing the pytest_bdd_step_error hook, leaving the step's failed flag as False and causing all steps to appear as "passed" in the JSON output even when the test failed.

### CODE REVIEW

**Comment by hiSandog** [comment]
Capturing `pytest.fail()` in cucumber JSON is the right direction, but the edge case is hook ordering around step teardown. It would be good to include a scenario where `pytest.fail()` happens inside the step body and another where failure is raised from a fixture/teardown path, then assert the JSON has the failed status and useful message in both cases. That keeps the report format reliable for CI consumers.

File: src/pytest_bdd/scenario.py [hunk 1]
```diff
@@ -247,7 +247,7 @@ def _execute_step_function(
         # so that we can allow "yield" statements in it
         return_value = call_fixture_func(fixturefunc=context.step_func, request=request, kwargs=kwargs)
 
-    except Exception as exception:
+    except BaseException as exception:
```
Comment (youtux) [inline src/pytest_bdd/scenario.py]: How about instead adding the catch for Failed instead?

```suggestion
    except (Exception, Failed) as exception:
```

**Review by youtux — CHANGES_REQUESTED** [review]

File: src/pytest_bdd/scenario.py (hunk 1 above)
Comment (jackhenderson101) [reply src/pytest_bdd/scenario.py]: Have narrowed the exception clause to only Exception and pytest.Failed

**Review by youtux — APPROVED** [review]
